test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,282 @@
+      });
+
+      it('findOneAndReplace returns queryPlanner explain result', async function () {
+        const explanation = await collection.findOneAndReplace(
+          { a: 1 },
+          { a: 2 },
+          { explain: 'queryPlanner' }
+        );
+        expect(explanation).property('queryPlanner').to.exist;
+      });
+    });
+
+    context('when explain is set to executionStats', () => {
+      it('executionStats property is used in explain result', async function () {
+        const explanation = await collection.deleteMany({ a: 1 }, { explain: 'executionStats' });
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+        expect(explanation).to.not.have.nested.property('executionStats.allPlansExecution');
+      });
+
+      it('distinct returns executionStats explain result', async function () {
+        const explanation = await collection.distinct('a', {}, { explain: 'executionStats' });
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+      });
+
+      it('find returns executionStats explain result', async function () {
+        const [explanation] = await collection
+          .find({ a: 1 }, { explain: 'executionStats' })
+          .toArray();
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+      });
+
+      it('findOne returns executionStats explain result', async function () {
+        const explanation = await collection.findOne({ a: 1 }, { explain: 'executionStats' });
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+      });
+    });
+
+    context('when explain is set to allPlansExecution', () => {
+      it('allPlansExecution property is used in explain result', async function () {
+        const explanation = await collection.deleteOne({ a: 1 }, { explain: 'allPlansExecution' });
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+        expect(explanation).nested.property('executionStats.allPlansExecution').to.exist;
+      });
+
+      it('find returns allPlansExecution explain result specified on cursor', async function () {
+        const explanation = await collection.find({ a: 1 }).explain('allPlansExecution');
+        expect(explanation).property('queryPlanner').to.exist;
+        expect(explanation).property('executionStats').to.exist;
+      });
+    });
+  });
+
+  it('find returns queryPlanner explain result', async () => {
+    const [explanation] = await collection.find({ a: 1 }, { explain: true }).toArray();
+    expect(explanation).property('queryPlanner').to.exist;
+  });
+
+  it('should honor boolean explain with aggregate', async function () {
```

Comment:
The aggregate tests provide another opportunity to use a context block for organization and getting rid of duplicate information in the test title and test setup.